sql/catalyst/src/main/scala/org/apache/spark/sql/types/NullType.scala

```diff
@@ -38,4 +38,12 @@ class NullType private() extends DataType {
  * @since 1.3.0
  */
 @Stable
-case object NullType extends NullType
+case object NullType extends NullType {
+
+  def containsNullType(dt: DataType): Boolean = dt match {
```

Comment:
let's not add a new method to a stable public class. Can we put it in the method `failNullType`?

Reply:
done